.github/workflows/ci.yml

```diff
@@ -45,4 +46,79 @@ jobs:
       run: poetry install --no-interaction
     
     - name: Test
-      run: poetry run pytest
\ No newline at end of file
+      run: poetry run pytest
+
+  build:
+    name: Build wheels
+    
+    needs: [test]
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+    
+    - name: Cache
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pypoetry
+        key: ${{ runner.os }}-python-${{ matrix.python-version }}-poetry-${{ hashFiles('**/poetry.lock') }}
+        restore-keys: ${{ runner.os }}-poetry-
+    
+    - name: Install Poetry
+      run: |
+        curl -sSL https://install.python-poetry.org | python3 -
+        echo "$HOME/.poetry/bin" >> $GITHUB_PATH
+
+    - name: Build
+      run: poetry build
+
+    - name: Upload wheels
+      uses: actions/upload-artifact@v2
+      with:
+        path: dist/*.*
+
+  publish:
+    name: Upload if release
+
+    needs: [build]
+
+    runs-on: ubuntu-latest
+    
+    if: github.event_name == 'release' && github.event.action == 'published'
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    - name: Cache
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pypoetry
+        key: ${{ runner.os }}-python-${{ matrix.python-version }}-poetry-${{ hashFiles('**/poetry.lock') }}
```

Comment:
question out of interest: why is the python version required here? should this not be version independent since its pure python?

Reply:
e.g. numpy is not pure-python and might cause problems. Not sure if this is responsable for the python 3.10 error. Needs some further investigations.